tests/aio.rs

```diff
@@ -0,0 +1,226 @@
+    /// identifier.
+    list: Box<[*mut libc::aiocb]>,
+    sev: libc::sigevent,
+}
+
+#[cfg(target_os = "freebsd")]
+impl Liocb {
+    fn listio(&mut self) -> Result<(), i32> {
+        unsafe {
+            let r = libc::lio_listio(
+                libc::LIO_NOWAIT,
+                self.list.as_ptr(),
+                self.list.len() as i32,
+                &mut self.sev as *mut libc::sigevent,
+            );
+            match r {
+                0 => Ok(()),
+                _ => Err(*libc::__error()),
+            }
+        }
+    }
+
+    fn new(inputs: Vec<Aiocb>) -> Liocb {
+        let mut aiocbs = inputs.into_boxed_slice();
+        for aiocb in aiocbs.iter_mut() {
+            aiocb.0.aio_lio_opcode = libc::LIO_WRITE;
+        }
+        let list = aiocbs
+            .iter_mut()
+            .map(|aiocb| &mut *aiocb.0 as *mut libc::aiocb)
+            .collect::<Vec<_>>()
+            .into_boxed_slice();
+        let sev = unsafe { mem::zeroed::<libc::sigevent>() };
+        Liocb {
+            _aiocbs: aiocbs,
+            list,
+            sev,
+        }
+    }
+}
+
+#[cfg(target_os = "freebsd")]
+impl Source for Liocb {
+    fn register(
+        &mut self,
+        registry: &Registry,
+        token: Token,
+        interests: Interest,
+    ) -> io::Result<()> {
+        assert!(interests.is_lio());
+        let udata = usize::from(token);
+        let kq = registry.as_raw_fd();
+        self.sev.sigev_notify = libc::SIGEV_KEVENT;
+        self.sev.sigev_signo = kq;
+        self.sev.sigev_value.sival_ptr = udata as *mut libc::c_void;
+        self.sev.sigev_notify_thread_id = 0;
+        Ok(())
+    }
+
+    fn reregister(
+        &mut self,
+        registry: &Registry,
+        token: Token,
+        interests: Interest,
+    ) -> io::Result<()> {
+        self.register(registry, token, interests)
+    }
+
+    fn deregister(&mut self, _registry: &Registry) -> io::Result<()> {
+        self.sev.sigev_notify = libc::SIGEV_NONE;
+        self.sev.sigev_value.sival_ptr = ptr::null_mut();
+        Ok(())
+    }
+}
+
+mod aio {
+    use super::*;
+
+    #[test]
+    fn smoke() {
+        let mut poll = Poll::new().unwrap();
+        let mut events = Events::with_capacity(8);
+
+        let f = tempfile().unwrap();
+        let mut aiocb = Aiocb::from_fd(f.as_raw_fd());
+        poll.registry()
+            .register(&mut aiocb, UDATA, Interest::AIO)
+            .unwrap();
+
+        expect_no_events(&mut poll, &mut events);
+        aiocb.fsync().unwrap();
+        expect_events(
+            &mut poll,
+            &mut events,
+            vec![ExpectEvent::new(UDATA, Interest::AIO)],
+        );
+    }
+}
+
+#[cfg(target_os = "freebsd")]
+mod lio {
+    use super::*;
+
+    #[test]
+    fn smoke() {
+        let data = b"hello, world!\n";
+        let mut poll = Poll::new().unwrap();
+        let mut events = Events::with_capacity(8);
+
+        let f0 = tempfile().unwrap();
+        let f1 = tempfile().unwrap();
```

Comment:
Nit: same as above, can you replace this with `temp_file("lio::smoke0/1")` from the `test::util` module?